### PR TITLE
chore: Deprecate show_native_filters

### PR DESCRIPTION
**Pull request by john-bodley**
### SUMMARY

This PR addresses recommendation #&#x2060;3 in https://github.com/apache/superset/issues/14383 by deprecating the `show_native_filters` dashboard property.

### ADDITIONAL INFORMATION

- [ ] Has associated issue: https://github.com/apache/superset/issues/14383
- [ ] Required feature flags:
- [ ] Changes UI
- [x] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [x] Migration is atomic, supports rollback & is backwards-compatible
  - [x] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/dashboard/components/DashboardBuilder/state.ts [hunk 1]
```diff
@@ -30,11 +30,6 @@ import {
 // eslint-disable-next-line import/prefer-default-export
 export const useNativeFilters = () => {
   const [isInitialized, setIsInitialized] = useState(false);
-  const showNativeFilters = useSelector<RootState, boolean>(
-    state =>
-      (getUrlParam(URL_PARAMS.showFilters) ?? true) &&
```
Comment (michael-s-molina) [inline superset-frontend/src/dashboard/components/DashboardBuilder/state.ts:35]: Can you remove `showFilters` from `URL_PARAMS` or use JSDoc [@deprecated](https://jsdoc.app/tags-deprecated.html)?

**Review by michael-s-molina — APPROVED** [review]
LGTM

**Comment by john-bodley** [comment]
@ktmud or @rusackas would you mind also stamping this? I'm not entirely sure why @michael-s-molina's approval didn't register.

**Comment by michael-s-molina** [comment]
> @ktmud or @rusackas would you mind also stamping this? I'm not entirely sure why @michael-s-molina's approval didn't register.

It's because of code ownership of Cypress files:
https://github.com/apache/superset/blob/473a69a7048719d706307fe7d6906026e68d427f/.github/CODEOWNERS#L25

**Review by rusackas — APPROVED** [review]
Cypress change looks fine, approving as codeowner.

**Comment by eschutho** [comment]
I just marked this as `risk:breaking-change` because it changes the api. Our automated tests broke, which flagged this PR. It looks like it's not included in 2.1, so we can just sneak this into 3.0.

**Pull request merged**